Build is failing on mac and linux while restoring nuget packages for two tests:

```
[xUnit.net 00:05:13.78]     Templates.Test.EmptyWebTemplateTest.EmptyWebTemplateCSharp [FAIL]

 | [11.662s] Templates.Test.EmptyWebTemplateTest Information: The template "ASP.NET Core Empty" was created successfully.
 | [11.673s] Templates.Test.EmptyWebTemplateTest Information: 
 | [11.673s] Templates.Test.EmptyWebTemplateTest Information: Processing post-creation actions...
 | [11.674s] Templates.Test.EmptyWebTemplateTest Information: Running 'dotnet restore' on /datadisks/disk1/work/9C2008E7/w/9F50091A/e/Templates/BaseFolder/AspNet.4jel0wrqdam/AspNet.4jel0wrqdam.csproj...
 | [311.903s] Templates.Test.EmptyWebTemplateTest Information: [ERROR] Restore failed.
```

```
[xUnit.net 00:11:59.26]     Templates.Test.WebApiTemplateTest.WebApiTemplateCSharp_IdentityWeb_BuildsAndPublishes(auth: "SingleOrg", args: null) [FAIL]

 | [27.729s] Templates.Test.WebApiTemplateTest Information: The template "ASP.NET Core Web API" was created successfully.
 | [27.729s] Templates.Test.WebApiTemplateTest Information: 
 | [27.729s] Templates.Test.WebApiTemplateTest Information: Processing post-creation actions...
 | [27.730s] Templates.Test.WebApiTemplateTest Information: Running 'dotnet restore' on /datadisks/disk1/work/9C2008E7/w/9F50091A/e/Templates/BaseFolder/AspNet.kthak30aajf/AspNet.kthak30aajf.csproj...
 | [328.501s] T
```

It looks like restore is interrupted after 5 minutes, but there is no log about the reason.

The SDK code that executes this step is 

https://github.com/dotnet/templating/blob/main/src/Microsoft.TemplateEngine.Cli/PostActionProcessors/DotnetRestorePostActionProcessor.cs#L92-L93

When run locally, the ProjectTemplates tests seem blocked, so the CI might actually have some sort of timeout that interrupts them.

--- a/src/ProjectTemplates/Shared/Project.cs
+++ b/src/ProjectTemplates/Shared/Project.cs
@@ -109,8 +109,6 @@ namespace Templates.Test.Helpers
                     Directory.Delete(TemplateOutputDir, recursive: true);
                 }
                 
-                // Temporary while investigating why this process occasionally never runs or exits on Debian 9
-                environmentVariables.Add("COREHOST_TRACE", "1");
                 using var execution = ProcessEx.Run(Output, AppContext.BaseDirectory, DotNetMuxer.MuxerPathOrDefault(), argString, environmentVariables);
                 await execution.Exited;
                 return new ProcessResult(execution);

